tests/conftest.py

```diff
@@ -30,6 +30,26 @@ def backing_h5ad(tmp_path):
     return tmp_path / "test.h5ad"
 
 
+@pytest.fixture(
+    params=[("h5ad", None), ("zarr", 2), ("zarr", 3)], ids=["h5ad", "zarr2", "zarr3"]
+)
+def diskfmt(request):
+    if (fmt := request.param[0]) == "h5ad":
+        yield fmt
+    else:
+        with ad.settings.override(zarr_write_format=request.param[1]):
+            yield fmt
+
+
+@pytest.fixture
+def diskfmt2(diskfmt):
+    if diskfmt == "h5ad":
+        with ad.settings.override(zarr_write_format=2):
+            yield "zarr"
+    else:
+        yield "h5ad"
```

Comment:
So diskfmt2 is supposed to be “the other one”?

I thought we tested the cartesian product for a reason, i.e. `test_thing[h5ad,h5ad]`, `test_thing[h5ad,zarr]`, `test_thing[zarr,h5ad]`, `test_thing[zarr,zarr]` and not just `test_thing[h5ad,zarr]`, `test_thing[zarr,h5ad]`.

But maybe we test that stuff elsewhere and don’t actually need to test it here, just want to make sure.

Reply:
yeah, if that’s the case, doing the same one twice doesn‘t test for anything new. OK!